llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1981,25 +1977,27 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
     return MemoryDepChecker::Dependence::Unknown;
   }
 
-  uint64_t TypeByteSize = DL.getTypeAllocSize(ATy);
-  bool HasSameSize =
-      DL.getTypeStoreSizeInBits(ATy) == DL.getTypeStoreSizeInBits(BTy);
-  if (!HasSameSize)
-    TypeByteSize = 0;
+  TypeSize AStoreSz = DL.getTypeStoreSize(ATy),
+           BStoreSz = DL.getTypeStoreSize(BTy);
+
+  // If store sizes are not the same, set TypeByteSize to zero, so we can check
+  // it in the caller.
```

Comment:
What caller?

You could also document that users must check for 0:
> If store sizes are not the same, set TypeByteSize to zero. Users of TypeByteSize much check for zero before assuming its validity.

Reply:
Side note: this is hopefully temporary, and I'm planning to iterate on https://github.com/llvm/llvm-project/pull/122318.